src/topology.jl

```diff
@@ -33,14 +49,41 @@ function Cart_create(comm_old::Comm, dims::AbstractArray{T,N}, periods::Array{T,
 end
 
 """
-    Cart_get(comm::Comm, maxdims::Integer)
+    rank = Cart_rank(comm::Comm, coords)
+
+Determine process rank in communicator `comm` with Cartesian structure.
+The `coords` array specifies the Cartesian coordinates of the process.
 
-Obtain information on the Cartesian topology of dimension `maxdims` underlying the 
-communicator `comm`. This is specified by two `Cint` arrays of `maxdims` elements
+# External links
+$(_doc_external("MPI_Cart_rank"))
+"""
+function Cart_rank(comm::Comm, coords::MPIBuffertype{Cint})
+    rank = Ref{Cint}()
+    # int MPI_Cart_rank(MPI_Comm comm, const int coords[], int *rank)
+    @mpichk ccall((:MPI_Cart_rank, libmpi), Cint,
+                  (MPI_Comm, Ptr{Cint}, Ptr{Cint}),
+                  comm, coords, rank)
+    Int(rank[])
+end
+
+function Cart_rank(comm::Comm, coords::AbstractArray{T}) where {T <: Integer}
+    ccoords = Cint.(coords[:])
```

Comment:
Why not just?
```suggestion
    ccoords = Cint.(coords)
```

Reply:
I agree, I just saw the same pattern used in other parts of the code (in `topology.jl` anyway), so I copied it. I'll change that one and a few other occurrences.

Reply:
It might be good to actually define it as
```
ccoords = [Cint(c) for c in coords]
```
and relax the type signature so that we can feed in tuples.

Reply:
Right. I guess we could do the same with `Dims_create!` and `Cart_create`, no? In their current version they don't accept tuples:
```julia
Dims_create!(nnodes::Integer, dims::AbstractArray{T,N}) where {T<:Integer, N}
Cart_create(comm_old::Comm, dims::AbstractArray{T,N}, periods::Array{T,N}, reorder) where T <: Integer where N
```

Reply:
By the way, is there any reason to accept `AbstractArray`s of arbitrary dimension in these functions? For me, it would only make sense for the `dims` argument to be an `AbstractVector` or a tuple...

I say this, because the current version of `Dims_create!` has the line
```julia
cdims = Cint.(dims[:])
```
Replacing this by `cdims = [Cint(d) for d in dims]` would make things fail if `dims` were a matrix. And that is actually the case in `test/test_cart_create.jl`, where `dims` is initialised as `[0 0 0]` (a 1×3 matrix).

Reply:
> Replacing this by cdims = [Cint(d) for d in dims] would make things fail if dims were a matrix.

That should still work (you would get a matrix back, but that's fine to pass to a `ccall`).